### DISCUSSION TITLE
Delete menu shown on Select all

### DISCUSSION THREAD

**Discussion by robbedekien** [laravel/nova-issues#4474] [Q&A] [answered]
My client reported a bug today that they could delete resources even if it isn't allowed

Technical situation:
I have a resource with a BelongsToMany relationship. Detaching the related models is **not allowed**.
When selecting all resources, suddenly the detach (delete) menu shows and the user receives the option to detach all related resources.

If I dig into the code, it seems like this is intended behaviour. My question is **why** the policy about detaching the resource is ignored when all resources are selected.

```
    /**
     * Determine whether the user is authorized to perform actions on the delete menu
     */
    canShowDeleteMenu() {
      return Boolean(
        this.authorizedToDeleteSelectedResources ||
          this.authorizedToForceDeleteSelectedResources ||
          this.authorizedToRestoreSelectedResources ||
          this.selectAllMatchingChecked
      )
    },
```

**Comment by crynobone** [comment, marked as answer]
You can only request to delete all, however only right before the deletion process we verify if the user has authorization once the model has been hydrated. If we do this during the selection on thousands if not millions of records your it would require huge resources on your server each time a users does select all.

**Reply by robbedekien** [reply]
Ok that makes a lot of sense. However imho showing the deletion menu is bad UX as the end-user doesn't understand what is happening in the background.
Would it be a solution to provide an optional `detachAll{Model}` (default permission Allowed) that just prevents the deletion model to be shown on select all?

This is just a suggestion as I understand the technical decision for showing the menu.

**Reply by VitaliyMinenko** [reply]
It posiible te remove SelectAll option from the top-bar? Just still select on this page option.
